doc/devel/conduct/code_of_conduct.rst

```diff
@@ -0,0 +1,167 @@
+Matplotlib Code of Conduct
+==========================
+
+
+Introduction
+------------
+
+This code of conduct applies to all spaces managed by the Matplotlib project,
+including all public and private mailing lists, issue trackers, wikis, blogs,
+Twitter, and any other communication channel used by our community.  The
+Matplotlib project does not currently organise in-person events, however events
+related to our community should have a code of conduct similar in spirit to
+this one.
+
+This code of conduct should be honored by everyone who participates in
+the Matplotlib community formally or informally, or claims any affiliation with
+the project, in any project-related activities and when representing the
+project, in any role.
+
+This code is not exhaustive nor complete. It serves to distill our common
+understanding of a collaborative, shared environment and goals. Please try to
+follow this code in spirit as much as in letter, to create a friendly and
+productive environment that enriches the surrounding community.
+
+
+Specific Guidelines
+-------------------
+
+We strive to:
+
+1. Be open. We invite anyone to participate in our community. We prefer to use
+   public methods of communication for project-related messages, unless
+   discussing something sensitive. This applies to messages for help or
+   project-related support, too; not only is a public support request much more
+   likely to result in an answer to a question, it also ensures that any
+   inadvertent mistakes in answering are more easily detected and corrected.
+
+2. Be empathetic, welcoming, friendly, and patient. We work together to resolve
+   conflict, and assume good intentions. We may all experience some frustration
+   from time to time, but we do not allow frustration to turn into a personal
+   attack. A community where people feel uncomfortable or threatened is not a
+   productive one.
+
+3. Be collaborative. Our work will be used by other people, and in turn we will
+   depend on the work of others. When we make something for the benefit of the
+   project, we are willing to explain to others how it works, so that they can
+   build on the work to make it even better. Any decision we make will affect
+   users and colleagues, and we take those consequences seriously when making
+   decisions.
+
+4. Be inquisitive. Nobody knows everything! Asking questions early avoids many
+   problems later, so we encourage questions, although we may direct them to
+   the appropriate forum. We will try hard to be responsive and helpful.
+
+5. Be careful in the words that we choose.  We are careful and respectful in
+   our communication and we take responsibility for our own speech. Be kind to
+   others. Do not insult or put down other participants.  We will not accept
+   harassment or other exclusionary behaviour, such as:
+
```

Comment:
In the sprit of collaboration and diversity of backgrounds something down these lines could be added:

Be considerate of others' participation.  Everyone should have an opportunity to be heard. Be patient and respectful of the fact that English is a second (or third or fourth!) language for some participants.